internal/expr/complete.go

```diff
@@ -42,8 +55,22 @@ func (pe *PreparedExpr) Complete(args ...any) (ce *CompletedExpr, err error) {
 		if !ok {
 			return nil, fmt.Errorf(`type %s not found, have: %s`, in.typ.String(), strings.Join(typeNames, ", "))
 		}
-		named := sql.Named("sqlair_"+strconv.Itoa(i), v.FieldByIndex(in.field.index).Interface())
-		qargs = append(qargs, named)
+
+		switch in.field.(type) {
+		case field:
+			f := in.field.(field)
+			named := sql.Named("sqlair_"+strconv.Itoa(i), v.FieldByIndex(f.index).Interface())
+			qargs = append(qargs, named)
+		case mapKey:
+			m := in.field.(mapKey)
+			if v.MapIndex(reflect.ValueOf(m.name)) == reflect.Value(reflect.ValueOf(nil)) {
```

Comment:
After line 38,  could do something like:

```Go
if _, ok = arg.(map[string]any); !ok {
     return nil, fmt.Errorf("not a map type with string keys")
} 
```
Using the function `CheckValidMapType()` right after line 38, so that we could also make sure that the map's name is `M` specifically as well. If we don't mind the map's name, I guess we can do a type assertion there. Another advantage I found with that function call is the ability to return different error messages about what's wrong with the input map, it not being a map, or the name not being `M`, or the key not being a string. 

If we still care about the map name to be `M`, but don't necessarily need a very specific error message there, then we could alternatively use the new function `IsValidMType()` defined in `mapinfo.go` as well. If we do that, `CheckValidMapType()` would no longer be needed, and can be removed.

Because `tv` carries reflection values for the Structs as well, to have the processing pipeline unchanged, I used reflections for the maps as well. I'll try to separate the logic in the coming days and see if it goes well. My first try today did not end well :')

Reply:
I have now changed the processing for the M type in `Complete()` - `tv` no longer keeps reflection for the M type.